README.md

```diff
@@ -50,3 +50,86 @@ To run over all the files in the repo manually:
+`--cmake-args -DENABLE_TRACING=OFF`.
+Example with colcon:
+```bash
+$ colcon build --packages-select abb_hardware_interface --cmake-args -DENABLE_TRACING=OFF
+```
+
+## Tracing with ros2_tracing
+
+Once built, you should start a tracing session with `ros2_tracing`:
+```bash
+ros2 trace --session-name abb-hardware-interface -u hardware_interface:* --list
+```
+The arguments are as follows:
+- `--session-name` is the name of the session. This is then reflected in the directory where the trace data is stored (e.g. `~/.ros/tracing/abb-hardware-interface`).
+- `-u hardware_interface:*` will trace all events from the `hardware_interface` namespace.
+- `--list` will list all the events that can be traced.
+
+After that, run the ROS 2 control nodes as usual.
+
+## Inspecting the trace data
+
+To inspect the data of the tracing session, we recommend using `babeltrace2`.
+If created you traced the example above, you can inspect the data with:
+```bash
+babeltrace2 ~/.ros/tracing/abb-hardware-interface | less
+```
+
+This will open a pager with the trace data. It will look something like this:
+
+```
+[14:51:33.967204443] (+0.000080284) sunrise-rtc hardware_interface:write_start: { cpu_id = 0 }, { vpid = 1519705, procname = "ros2_control_no", vtid = 1519779 }, { if_name = "ABBMultiInterfaceHardwareSmall" }
+[14:51:33.967218440] (+0.000013997) sunrise-rtc hardware_interface:write_end: { cpu_id = 0 }, { vpid = 1519705, procname = "ros2_control_no", vtid = 1519779 }, { if_name = "ABBMultiInterfaceHardwareSmall" }
+[14:51:33.967219471] (+0.000001031) sunrise-rtc hardware_interface:write_start: { cpu_id = 0 }, { vpid = 1519705, procname = "ros2_control_no", vtid = 1519779 }, { if_name = "ABBMultiInterfaceHardwareBig" }
+[14:51:33.967221196] (+0.000001725) sunrise-rtc hardware_interface:write_end: { cpu_id = 0 }, { vpid = 1519705, procname = "ros2_control_no", vtid = 1519779 }, { if_name = "ABBMultiInterfaceHardwareBig" }
+[14:51:33.971103015] (+0.003881819) sunrise-rtc hardware_interface:read_start: { cpu_id = 0 }, { vpid = 1519705, procname = "ros2_control_no", vtid = 1519779 }, { if_name = "ABBMultiInterfaceHardwareSmall" }
+[14:51:33.971147404] (+0.000044389) sunrise-rtc hardware_interface:read_end: { cpu_id = 0 }, { vpid = 1519705, procname = "ros2_control_no", vtid = 1519779 }, { if_name = "ABBMultiInterfaceHardwareSmall" }
+[14:51:33.971148861] (+0.000001457) sunrise-rtc hardware_interface:read_start: { cpu_id = 0 }, { vpid = 1519705, procname = "ros2_control_no", vtid = 1519779 }, { if_name = "ABBMultiInterfaceHardwareBig" }
+[14:51:33.971154327] (+0.000005466) sunrise-rtc hardware_interface:read_end: { cpu_id = 0 }, { vpid = 1519705, procname = "ros2_control_no", vtid = 1519779 }, { if_name = "ABBMultiInterfaceHardwareBig" }
+```
+
+In this example we have two controllers, `ABBMultiInterfaceHardwareSmall` and `ABBMultiInterfaceHardwareBig`.
+The events `write_start`, `write_end`, `read_start`, and `read_end` events are logged for both controllers.
```

Comment:
No comma before and.

Reply:
[Oxford comma](https://en.wikipedia.org/wiki/Serial_comma), baby :smirk: